### PR TITLE
Rework achievements system

### PR DESCRIPTION
**Pull request by jabsatz**
Hey @5FB5, here's the PR I promised.

Please take your time to review this. I realize these are big changes, so please don't hesitate to ask for corrections or questions about my code. I'd be happy to answer/fix it.

The only thing that's missing is a way to make the json file backwards compatible, I can probably write a script to update it if a previous user jumps to the newest version, and I can also write a migration guide for this update, but first I'd like to see what you think.

# Reordering project:
- Moved most of the functionality from the "AchievementSystem" node to the "Global" node, except the things related with rendering the notifications
- Moved the "AchievementData" functionality to the "Global" node and deleted "AchievementData".
- Renamed the "Global" component to "AchievementManager" and the "AchievementSystem" node to "AchievementNotifications"
- I also deleted the "getter" functions for each achievement's field, and replaced them with a `getAchievement(id)` function that returns the whole dictionary, since it seemed unnecessary to have so many getters.
- A new folder `gd-notifications` was created in the root. This is standard practice for plugins like Dialogic, where a separate folder with all the customizable files is included so the users do not have to modify anything within the `addons/gd-notifications` folder. That way, if a user wants to upgrade, they don't have to worry about changes they made in that folder. The new folder includes:
  - The achievements.json file
  - achievement_notification.tscn and .gd files, which represent a single notification (in case the user wants to make any visual change to how the notification looks)
- Converted the plugin into a godot project with a `demo` folder, with a scene that showcases basic usage of the plugin. This can usually clear a lot of doubts if for any reason the README is not enough
- Also updated the README 😃 

# New features
- The global "AchievementManager" is now added automatically when enabling the plugin
- The "AchievementManager" now has 2 functions to track achievements: `progress_achievement(id, progress)` which increases the progress of a specific achievement, and eventually unlocks it, and `unlock_achievement(id)` which instantly unlocks an achievement
  - From your code it doesn't look like you knew this, but in godot if you have an Autoload, you do not need to use `get_node('/root/AchievementManager').unlock_achievement(id)`, you can simply call `AchievementManager.unlock_achievement(id)`, which is very user friendly
  - These functions emit a new signal `achievement_unlocked(achievement)`. The "AchievementNotifications" will automatically receive these and display them on-screen.
- The "AchievementNotifications" can now stack in any direction (before, they only stacked down) and from any corner of the screen. These can be set by a property in the editor.
- The single `gd-achievements/achievement_notification.tscn` is now standalone and can be customized by the user without worries (this is called "inversion of control", where instead of giving the user a lot of properties to change on a set component, we let them define the component with a couple of methods we will later use. full description in the README)
- The shape of "achievements.json" is changed, as there were some typos and the "progress" didn't make sense to be at 0 if the achievement doesn't exist. If there is progress, there will also be a "goal". The progress will increase when using `AchievementManager.progress_achievement()` until it reaches the goal, then it'll emit the signal to show the achievement. Now it looks like this:
```
{
  "infinite_suffering": {
    "name": "Infinite Suffering",
    "description": "Die 5 times",
    "is_secret": false,
    "goal": 5,
    "current_progress": 0,
    "icon_path": "res://gd-achievements/achievement_test.png",
    "achieved": false
  },
  "headache": {
    "name": "You'll have a headache in the morning",
    "description": "Kill 4 monsters with a single explosion",
    "is_secret": false,
    "icon_path": "res://gd-achievements/achievement_test.png",
    "achieved": false
  }
}
```

### CODE REVIEW

File: default_env.tres [hunk 1]
```diff
@@ -0,0 +1,7 @@
+[gd_resource type="Environment" load_steps=2 format=2]
```
Comment (5FB5) [inline default_env.tres:1]: Do we need a .tres file ?

**Review by 5FB5 — CHANGES_REQUESTED** [review]
I'm thinking about backwards compatible, maybe it will be cool to check old achievements.json file for old structure and just replace old data to the new structure. Maybe it will be more comfortable to make it in another python (like importer) script or in generator script to reduce problems for old users that will update his plugin

**Comment by 5FB5** [comment]
But in other, cool update! :)

**Comment by jabsatz** [comment]
Yes, I agree, thanks for looking over the changes. I'll figure out how to make the .py script update the achievements file, or create a new one (I'm not terribly familiar with python, but looking over your code I think I can figure it out).

In the future, it'd also be possible to create an in-engine editor for the achievements, as other plugins do. I have limited experience with that, but I'd be glad to give it a shot (not proposing this for now, only after this whole PR is done)

A couple of examples of what I'm talking about:

This is SoundManager, the tab it creates helps users set volume/soundmaps
image

This is Dialogic, you can create dialogues between characters on it
image

(this is just to show what's possible, for this plugin the editor would be much simpler)

**Comment by 5FB5** [comment]
Nice idea with in-engine editor!

**Comment by jabsatz** [comment]
Made requested changes and added upgrade guide at the end of the README

**Review by 5FB5 — COMMENTED** [review]
Great!

**Review by 5FB5 — APPROVED** [review]

**Comment by jabsatz** [comment]
Thanks for approving 😄 

After merging, make sure to bump the plugin and script version as you see fit, I'm not sure what versioning convention you use so I didn't touch any of them

**Comment by 5FB5** [comment]
Later I'll update info for Godot Asset site and other, py script's info updated too))

**Pull request merged**